packages/@lwc/engine/src/framework/base-lightning-element.ts

```diff
@@ -232,18 +228,26 @@ function BaseLightningElementConstructor(this: LightningElement): LightningEleme
 
 BaseLightningElementConstructor.prototype = {
     constructor: BaseLightningElementConstructor,
-    dispatchEvent() {
-        const elm = getLinkedElement(this);
-        // Typescript does not like it when you treat the `arguments` object as an array
-        // @ts-ignore type-mismatch;
-        return elm.dispatchEvent.apply(elm, arguments);
+
+    dispatchEvent(event: Event): boolean {
+        const {
+            elm,
+            renderer: { dispatchEvent },
+        } = getAssociatedVM(this);
```

Comment:
This pattern is repeated thought out this prototype. For a given component instance, will the renderer always be the same? If so, may be we can directly associate the renderer with the component instance and not have to look it up using the VM.
```suggestion
        const {
            elm,
            renderer: { dispatchEvent },
        } = getAssociatedVM(this);
```